be/src/io/async_flush_output_stream.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "async_flush_output_stream.h"
+
+namespace starrocks::io {
+
+int64_t AsyncFlushOutputStream::SliceChunk::append(const uint8_t* data, int64_t size) {
+    int64_t old_size = static_cast<int64_t>(buffer_.size());
+    int64_t to_write_bytes = std::min(max_size_ - old_size, size);
+    buffer_.resize(buffer_.size() + to_write_bytes);
```

Comment:
reserve?

Reply:
good idea. Let me see how to do it.

Reply:
use raw::RawVector to avoid zeroing overhead.